src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -101,64 +143,190 @@ function SearchRouter({onRouterClose}: SearchRouterProps) {
+                        .filter((chat) => chat.text?.toLowerCase()?.includes(autocompleteValue.toLowerCase()))
+                        .sort((chatA, chatB) => (chatA > chatB ? 1 : -1))
+                        .slice(0, 10);
+                    setAutocompleteSuggestions(filteredChats.map((chat) => ({text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.IN}:${chat.text}`, query: `${chat.reportID}`})));
+                    return;
+                }
+                case CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE: {
+                    const filteredTypes = typeAutocompleteList
+                        .filter((type) => type.toLowerCase().includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(type.toLowerCase()))
+                        .sort();
+                    setAutocompleteSuggestions(filteredTypes.map((type) => ({text: `${CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE}:${type}`, query: `${type}`})));
+                    return;
+                }
+                case CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS: {
+                    const filteredStatuses = statusAutocompleteList
+                        .filter((status) => status.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(status))
+                        .sort()
+                        .slice(0, 10);
+                    setAutocompleteSuggestions(filteredStatuses.map((status) => ({text: `${CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS}:${status}`, query: `${status}`})));
+                    return;
+                }
+                case CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE: {
+                    const filteredExpenseTypes = expenseTypes
+                        .filter((expenseType) => expenseType.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(expenseType))
+                        .sort();
+                    setAutocompleteSuggestions(
+                        filteredExpenseTypes.map((expenseType) => ({
+                            text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE}:${expenseType}`,
+                            query: `${expenseType}`,
+                        })),
+                    );
+                    return;
+                }
+                case CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID: {
+                    const filteredCards = cardAutocompleteList
+                        .filter((card) => card.toLowerCase().includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(card.toLowerCase()))
+                        .sort()
+                        .slice(0, 10);
+                    setAutocompleteSuggestions(
+                        filteredCards.map((card) => ({
+                            text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID}:${card}`,
+                            query: `${card}`,
+                        })),
+                    );
+                    return;
+                }
```

Comment:
We should have used the card last four as well for auto suggestion. Coming from https://github.com/Expensify/App/issues/57994